src/ucs/async/eventfd.c

```diff
@@ -0,0 +1,79 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023-2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "eventfd.h"
+
+#include <ucs/debug/log.h>
+#include <ucs/sys/sys.h>
+
+
+ucs_status_t ucs_async_eventfd_create(int *fd)
+{
+
+    *fd = eventfd(0, EFD_CLOEXEC | EFD_NONBLOCK);
+    if (*fd == -1) {
+        ucs_error("Failed to create event fd: %m");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+void ucs_async_eventfd_destroy(int fd)
+{
+    close(fd);
+}
+
+ucs_status_t ucs_async_eventfd_try_dummy_read(int fd)
+{
+    uint64_t dummy;
+    int ret;
+
+    ret = read(fd, &dummy, sizeof(dummy));
+    if (ret == sizeof(dummy)) {
+        return UCS_OK;
+    } else if (ret == -1) {
+        if (errno == EAGAIN) {
+            return UCS_ERR_NO_PROGRESS;
+        } else if (errno != EINTR) {
+            ucs_error("read interrupted: %m");
+            return UCS_ERR_IO_ERROR;
+        } else {
+            return UCS_OK;
+        }
+    } else {
+        ucs_assert(ret == 0);
+        return UCS_ERR_NO_PROGRESS;
+    }
+}
+
+ucs_status_t ucs_async_eventfd_try_dummy_write(int fd)
+{
+    uint64_t dummy = 1;
+    int ret;
+
+    do {
+        ret = write(fd, &dummy, sizeof(dummy));
+        if (ret == sizeof(dummy)) {
+            return UCS_OK;
+        } else if (ret == -1) {
+            if (errno == EAGAIN) {
+                continue;
+            } else if (errno != EINTR) {
+                ucs_error("write interrupted: %m");
+                return UCS_ERR_IO_ERROR;
+            }
+        } else {
+            ucs_assert(ret == 0);
+        }
```

Comment:
unify code with ucs_async_eventfd_try_dummy_read by helper function that gets read/write as callback